processing/data_collection/gazette/spiders/ap_macapa.py

```diff
@@ -0,0 +1,79 @@
+import datetime as dt
+
+import dateparser
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class ApMacapaSpider(BaseGazetteSpider):
+
+    name = "ap_macapa"
+    allowed_domains = ["macapa.ap.gov.br"]
+    start_date = None
+
+    TERRITORY_ID = "1600303"
+
+    def __init__(self, start_date=None, end_date=None, *args, **kwargs):
+        if start_date is None:
+            # Start from yesterday
+            start_date = dt.date.today() + dt.timedelta(days=-1)
+        else:
+            start_date = dateparser.parse(start_date)
+
+        if end_date is None:
+            # End today
+            end_date = dt.date.today()
+        else:
+            end_date = dateparser.parse(end_date)
+
+        self.start_date = start_date
+        self.end_date = end_date
```

Comment:
I'm considering the idea of moving all this logic of start and end date to the base class. Thus, we can try to add the same idea for other spiders, when possible. Please, update the PR.

Reply:
Done. The changes to the base class are proposed on #265.